### PR TITLE
Add logic for CDXJ generation

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made? 🤔
Ref #449 

## How was this change tested? 🤨
Test suite

### CODE REVIEW

**Review by edsu — COMMENTED** [review]
I know that @lwrubel is looking at this too, so I'll just drop my comment.

File: spec/robots/dor_repo/was_crawl_dissemination/cdxj_generator_spec.rb [hunk 1]
```diff
@@ -9,9 +9,23 @@
...
+      allow(Dor::WasCrawl::Dissemination::Utilities).to receive(:run_sys_cmd)
+      allow(Dor::WasCrawl::Dissemination::Utilities).to receive(:warc_file_list).and_return(['foo/number1.warc'])
+    end
 
-    it 'returns a skipped status' do
-      expect(perform.status).to eq('skipped')
+    it 'runs the cdxj-indexer' do
+      perform
+      expect(Dor::WasCrawl::Dissemination::Utilities).to have_received(:run_sys_cmd)
+        .with('cdxj-indexer /web-archiving-stacks/data/collections/xx123xx1234/dd/116/zh/0343/foo/number1.warc --output tmp/druid:dd116zh0343/number1.cdxj --sort --post-append 2>> log/cdx_indexer.log',
```
[the remark this answers is not in the document]
Comment (jcoyne) [reply spec/robots/dor_repo/was_crawl_dissemination/cdxj_generator_spec.rb:27]: Ensuring it's installed is the job of the capistrano configs. I'm testing the ruby code here. This is how we test all external programs in unit tests.

**Comment by jcoyne** [comment]
@edsu seeing that the indexer is deployed and working up will be covered by https://github.com/sul-dlss/infrastructure-integration-test/issues/434

**Review by lwrubel — APPROVED** [review]
Looks good.

**Pull request merged**